src/content/guides/upgrade-tiptap-v2.mdx

```diff
@@ -0,0 +1,308 @@
+
+Extensions
+- import Focus from '@tiptap/extension-focus'
+- import Placeholder from '@tiptap/extension-placeholder'
+- import History from '@tiptap/extension-history'
+- import Dropcursor from '@tiptap/extension-dropcursor'
+- import Gapcursor from '@tiptap/extension-gapcursor'
+- import CharacterCount from '@tiptap/extension-character-count'
++ import { Focus, Placeholder, History, Dropcursor, Gapcursor, CharacterCount } from '@tiptap/extensions'
+
+React Menus
+- import { BubbleMenu, FloatingMenu } from '@tiptap/react'
++ import { BubbleMenu, FloatingMenu } from '@tiptap/react/menus'
+
+Vue 3 Menus
+- import { BubbleMenu, FloatingMenu } from '@tiptap/vue-3'
++ import { BubbleMenu, FloatingMenu } from '@tiptap/vue-3/menus'
+
+Vue 2 Menus
+- import { BubbleMenu, FloatingMenu } from '@tiptap/vue-2'
++ import { BubbleMenu, FloatingMenu } from '@tiptap/vue-2/menus'
+```
+
+### Menu Systems
+
+Tippy.js has been replaced with Floating UI for all floating elements. This affects:
+
+- @tiptap/extension-floating-menu
+- @tiptap/extension-bubble-menu
+- @tiptap/extension-mention
+- @tiptap/suggestion
+
+Example migration:
+
+```tsx
+// Before
+import { BubbleMenu } from '@tiptap/react'
+
+const menu = <BubbleMenu tippyOptions={{ duration: 100 }}>{/* menu content */}</BubbleMenu>
+
+// After
+import { BubbleMenu } from '@tiptap/react/menus'
+import { offset } from '@floating-ui/dom'
+const menu = (
+  <BubbleMenu
+    options={{
+      middleware: [offset(6)],
+      placement: 'top',
+    }}
+  >
+    {/* menu content */}
+  </BubbleMenu>
+)
+```
+
+- New required dependency: `@floating-ui/dom`. Install it with:
+
+```bash
+npm install @floating-ui/dom@^1.6.0
+```
+
+### API Changes Text Style Updates
+
+- `mergeNestedSpanStyles` now defaults to `true`
+- New text style extensions added:
+  - `font-size`
+  - `background-color`
+  - `line-height`
+- `TextStyleKit` API available for configuring multiple extensions
+
+You can now use the new `TextStyleKit` for easier configuration:
+
+```ts
+import { TextStyleKit } from '@tiptap/extension-text-style'
+
+new Editor({
+  extensions: [
+    TextStyleKit.configure({
+      backgroundColor: {
+        types: ['textStyle'],
+      },
+      color: {
+        types: ['textStyle'],
+      },
+      fontFamily: {
+        types: ['textStyle'],
+      },
+      fontSize: {
+        types: ['textStyle'],
+      },
+      lineHeight: {
+        types: ['textStyle'],
+      },
+    }),
+  ],
+})
+```
+
+### Command Changes
+
+- `clearContent` and `setContent` now emit updates by default
+- `setContent` signature changed to (content, options)
+- `insertContent` behavior modified to prevent splitting text nodes at the beginning
+
+### NodeView Changes
+
+The `getPos` function in NodeViewRendererProps can now return `undefined`. Update your code to handle this case:
+
+```ts
+// Before
+const pos = nodeViewProps.getPos()
+
+// After
+const pos = nodeViewProps.getPos()
+if (pos !== undefined) {
+  // use pos
+}
```

Comment:
What does it mean the `getPos() === undefined`, and how should it be handled in that case?

Does it mean the node is not inserted yet / finished rendered?

Reply:
https://github.com/ProseMirror/prosemirror-view/pull/153

Once the node has been removed from the document, `getPos` is no longer valid (since React can hold a reference to `getPos` after something is deleted)

Reply:
ahh, okay, in my specific case, I have an action `addChild`, so if `getPos()` returns `undefined`, I should not do anything probably :) 

thanks for the quick reply!